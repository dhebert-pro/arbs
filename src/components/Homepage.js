import React from "react";
import { Link } from "routes";

const Homepage = ({ datas }) => <div>
    This is Homepage<br/>
    <Link route='/category/hello-world'><a>Aller sur la page hello-world</a></Link>
    {datas && datas.map(data => <div key={data.id}>{data.title} ***** {data.body}</div>)}
</div>;

export default Homepage;