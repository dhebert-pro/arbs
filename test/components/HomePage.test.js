/*global expect test describe:true */
/*eslint no-undef: "error"*/
import React from "react";
import { shallow } from "enzyme";
import Homepage from "components/Homepage";
import renderer from "react-test-renderer";

const datas = [
  {
    _id: "ID",
    nom: "NOM",
    adresse: "ADRESSE",
    tel: "TEL",
    description: "DESCRIPTION"
  }
];

describe("Homepage", () => {
  test("should be rendered", done => {
    const component = renderer.create(<Homepage datas={datas} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    done();
  });
  test("should render datas", done => {
    const component = shallow(<Homepage datas={datas} />);
    expect(component.length).toEqual(1);
    expect(component.find("div > div[name='datas']").text()).toEqual(
      "NOM ** ADRESSE ** TEL ** DESCRIPTION"
    );
    done();
  });
});
