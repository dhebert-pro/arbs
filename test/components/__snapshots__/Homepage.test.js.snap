// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage should be rendered 1`] = `
<div
  name="homepage"
>
  This is Homepage
  <br />
  <a
    href="/category/hello-world"
    onClick={[Function]}
  >
    Aller sur la page hello-world
  </a>
  <div
    name="datas"
  >
    <div>
      NOM
       ** 
      ADRESSE
       ** 
      TEL
       ** 
      DESCRIPTION
    </div>
  </div>
</div>
`;
